detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	return awsAuth != "", nil
+}
+
+// isK8s checks if the current environment is running in a Kubernetes environment
+func isK8s(utils detectorUtils) bool {
+	return utils.fileExists(k8sTokenPath) && utils.fileExists(k8sCertPath)
+}
+
+// fileExists checks if a file with a given filename exists
+// this function implements the detectorUtils interface
+func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// fetchString executes an HTTP request with a given HTTP Method and URL string
+// this function implements the detectorUtils interface
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+
+	// Create new HTTP request object
+	request, err := http.NewRequest(httpMethod, URL, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to create new HTTP request with method=%s, URL=%s", httpMethod, URL)
+	}
+
+	// Set HTTP request header with authentication credentials
+	authHeader, err := getK8sCredHeader()
+	if err != nil {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	// Get certificate
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file with path %s", k8sCertPath)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	// Set HTTP request timeout and add certificate
+	client := &http.Client{
+		Timeout: millisecondTimeOut * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	// Execute HTTP request
+	response, err := client.Do(request)
+	if err != nil {
+		return "", fmt.Errorf("failed to execute HTTP request with method=%s, URL=%s", httpMethod, URL)
+	}
+
+	// Retrieve response body from HTTP request
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response from HTTP request with method=%s, URL=%s", httpMethod, URL)
+	}
+
+	return string(body), nil
+}
+
+// getK8sCredHeader retrieves the kubernetes credential information
+func getK8sCredHeader() (string, error) {
+	content, err := ioutil.ReadFile(k8sTokenPath)
+	if err != nil {
+		return "", fmt.Errorf("getK8sCredHeader() error: cannot read file with path %s", k8sTokenPath)
+	}
+
+	return "Bearer " + string(content), nil
+}
+
+// getClusterName retrieves the clusterName resource attribute
+func getClusterName(utils detectorUtils) (string, error) {
+	resp, err := utils.fetchString("GET", k8sSvcURL+cwConfigmapPath)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: %s", err.Error())
+	}
+
+	// parse JSON object returned from HTTP request
+	var parsedResp JSONResponse
+	err = json.Unmarshal([]byte(resp), &parsedResp)
+	if err != nil {
+		return "", fmt.Errorf("getClusterName() error: cannot parse JSON %s", resp)
```

Comment:
```suggestion
		return "", fmt.Errorf("getClusterName() error: cannot parse JSON: %w", w)
```

Putting the entire response into the error message might be a bit much.